cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -109,4 +116,14 @@ protected TrackedTransform getTransformation(String pluginType, String stageName
     return new TrackedTransform<>(transform, stageMetrics, "aggregator.groups", TrackedTransform.RECORDS_OUT);
   }
 
+  protected static <IN, OUT> TrackedTransform<IN, OUT> getTrackedJoinOnStep(Transformation<IN, OUT> transform,
```

Comment:
this is the same as getTrackedGroupStep. We should just rename that method instead of adding a copy.